src/key-managment/AwsKeyManagement.cs

```diff
@@ -0,0 +1,107 @@
+using System;
+using System.IO;
+using System.Net;
+using System.Text;
+using System.Threading.Tasks;
+using Amazon.KeyManagementService;
+using Amazon.KeyManagementService.Model;
+
+namespace Kamus.KeyManagement
+{
+    public class AwsKeyManagement : IKeyManagement
+    {
+        private readonly IAmazonKeyManagementService mAmazonKeyManagementService;
+        private readonly SymmetricKeyManagement mSymmetricKeyManagement;
+        private readonly string mUserArn;
+
+        public AwsKeyManagement(IAmazonKeyManagementService amazonKeyManagementService, SymmetricKeyManagement symmetricKeyManagement, string userArn)
+        {
+            mAmazonKeyManagementService = amazonKeyManagementService;
+            mSymmetricKeyManagement = symmetricKeyManagement;
+            mUserArn = userArn;
+        }
+
+        public async Task<string> Encrypt(string data, string serviceAccountId, bool createKeyIfMissing = true)
+        {
+            var masterKeyAlias = $"alias/kamus/{KeyIdCreator.Create(serviceAccountId)}";
+            (var encryptionKey, var encryptedEncryptionKey ) = await GenerateEncryptionKey(masterKeyAlias);
+            mSymmetricKeyManagement.SetEncryptionKey(Convert.ToBase64String(encryptionKey.ToArray()));
+            var encryptedData = await mSymmetricKeyManagement.Encrypt(data, serviceAccountId);
+
+            return "env" + "$" + encryptedEncryptionKey + "$" + encryptedData;
+
+        }
+
+        public async Task<string> Decrypt(string encryptedData, string serviceAccountId)
+        {
+            var encryptedEncryptionKey = encryptedData.Split('$')[1];
+            var actualEncryptedData = encryptedData.Split('$')[2];
+            
+            var decryptionResult = await mAmazonKeyManagementService.DecryptAsync(new DecryptRequest
```

Comment:
You don't need to reference the relevant master key?

Reply:
no, it's part of the encrypted data metadata. (that's how to SDK works)

Reply:
magic